core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/DomainService.java

```diff
@@ -89,7 +104,37 @@ public List<Domain> findAll(Predicate<Domain> filter) {
 
   @PreAuthorize("hasPermission(#domain, 'DELETE')")
   public void delete(Domain domain) {
-    throw new UnsupportedOperationException("Domain deletion not currently supported.");
+    handlerService.handleDelete(domain);
+
+    processView
+      .findAll(p -> p.getKey().getDomainKey().equals(domain.getKey()))
+      .findAny()
+      .ifPresent(ex -> { throw errorHandler(); });
+
+    streamView
+      .findAll(s -> s.getKey().getDomainKey().equals(domain.getKey()))
+      .findAny()
+      .ifPresent(ex -> { throw errorHandler(); });
+
+    schemaView
+      .findAll(sc -> sc.getKey().getDomainKey().equals(domain.getKey()))
+      .findAny()
+      .ifPresent(ex -> { throw errorHandler(); });
+
+    producerView
+      .findAll(p -> p.getKey().getStreamKey().getDomainKey().equals(domain.getKey()))
+      .findAny()
+      .ifPresent(ex -> { throw errorHandler(); });
+
+    consumerView
+      .findAll(c -> c.getKey().getStreamKey().getDomainKey().equals(domain.getKey()))
+      .findAny()
+      .ifPresent(ex -> { throw errorHandler(); });
+
+    domainRepository.delete(domain);
   }
 
+  private static IllegalStateException errorHandler() {
+    return new IllegalStateException("Domain in use.");
```

Comment:
I'd add to the exception where the Domain is in use (`schema`, `process` etc)

Reply:
ok , i was thinking the same . But not implemented , i can't remember, what was in my mind. 
Adding it .